src/main/java/com/sleekbyte/tailor/format/Formatter.java

```diff
@@ -1,17 +1,37 @@
 package com.sleekbyte.tailor.format;
 
+import com.sleekbyte.tailor.common.ColorSettings;
+import com.sleekbyte.tailor.common.ExitCode;
 import com.sleekbyte.tailor.output.ViolationMessage;
 
+import java.io.File;
 import java.io.IOException;
 import java.util.List;
 
 /**
- * Interface used by the Printer class to display violation messages/errors.
- * New formats can be added by implementing this interface.
+ * Formatter used by the Printer class to display violation messages.
+ * New formats can be added by extending this abstract base class.
  */
-public interface Formatter {
+public abstract class Formatter {
 
-    public void displayViolationMessages(List<ViolationMessage> violationMessages) throws IOException;
+    protected File inputFile;
+    protected ColorSettings colorSettings;
+
+    public Formatter(File inputFile, ColorSettings colorSettings) {
+        this.inputFile = inputFile;
+        this.colorSettings = colorSettings;
+    }
+
+    public abstract void displayViolationMessages(List<ViolationMessage> violationMessages) throws IOException;
```

Comment:
Can we have brief method comments explaining the expected functionality of each method and what this method is used for (ex. `displaySummary` is used for printing the summary statistics once the full analysis of all files is complete).  This will make it easier for any new formatters that extend this class to implement these methods (ex. should `displayParseErrorMessage` handle the error message for one file, or should it print a statement saying that `x` number of files had a parsing error?)